modules/contact/src/actions/ContactAction.C

```diff
@@ -210,6 +213,23 @@ ContactAction::addMortarContact()
   }
 
   if (_current_task == "add_variable")
+  {
+    auto params = _factory.getValidParams("MooseVariable");
+    // determine necessary order
+    const bool second = _problem->mesh().hasSecondOrderElements();
+
+    params.set<MooseEnum>("order") = second ? "SECOND" : "FIRST";
+    params.set<MooseEnum>("family") = "LAGRANGE";
+
+    // Loop through the displacement variables
+    for (const auto & disp : displacements)
+    {
+      // Create displacement variables
```

Comment:
The displacement variables are created by the TM master action, no need to duplicate that functionality in the contact action

Reply:
Makes sense 👍 This is now resolved